types/address.go

```diff
@@ -571,6 +576,85 @@ func (ca ConsAddress) Format(s fmt.State, verb rune) {
 	}
 }
 
+// ----------------------------------------------------------------------------
+// EE address
+// ----------------------------------------------------------------------------
+//
+
+// EEAddress length is 32
+type EEAddress []byte
+
+func (ee EEAddress) Bytes() []byte {
+	return ee
+}
+
+func verifyEEAddress(src []byte) error {
+	if len(src) != 32 {
+		return errors.New("Incorrect address length")
+	}
+
+	return nil
+}
+
+// GetEEAddressFromCryptoPubkey converts blake2b 32 byte address from amino-encoded 33 bit secp256k1 public key
+func GetEEAddressFromCryptoPubkey(cryptoPubKey crypto.PubKey) (EEAddress, error) {
+	cdc := codec.New()
+	cryptoAmino.RegisterAmino(cdc)
+
+	var cont tmsecp256k1.PubKeySecp256k1
+	err := cdc.UnmarshalBinaryBare(cryptoPubKey.Bytes(), &cont)
+	if err != nil {
+		return nil, err
+	}
+
+	eeAddress := eeutil.Blake2b256(cont.Bytes())
+
+	err = verifyEEAddress(eeAddress)
+	if err != nil {
+		return nil, err
+	}
+
+	return eeAddress, nil
+}
+
+// MustGetEEAddressFromCryptoPubkey panics if there is error in parsing
+func MustGetEEAddressFromCryptoPubkey(cryptoPubKey crypto.PubKey) EEAddress {
+	res, err := GetEEAddressFromCryptoPubkey(cryptoPubKey)
+	if err != nil {
+		panic(err)
+	}
+	return res
+}
+
+// GetEEAddressFromBech32 converts from bech32-encoded public key
+func GetEEAddressFromBech32(bech32String string) (EEAddress, error) {
+	parsedPubKey, err := GetAccPubKeyBech32(bech32String)
+	if err != nil {
+		return nil, err
+	}
+
+	res, err := GetEEAddressFromCryptoPubkey(parsedPubKey)
+	if err != nil {
+		return nil, err
+	}
+
+	return res, nil
+}
+
+// MustGetEEAddressFromBech32 panics if there is error in parsing
+func MustGetEEAddressFromBech32(bech32String string) EEAddress {
+	res, err := GetEEAddressFromBech32(bech32String)
+	if err != nil {
+		panic(err)
+	}
+	return res
+}
+
+// GetEEAddressFromSecp256k1PubKey derives Black2b256-hashed address
+func GetEEAddressFromSecp256k1PubKey(pubkey tmsecp256k1.PubKeySecp256k1) EEAddress {
+	return eeutil.Blake2b256(pubkey.Bytes())
```

Comment:
This also implies possibility of panic or returning invalid output without errors.

Reply:
resolve since it is reported in #54